react-components/src/query/useSearchMappedEquipmentAssetMappings.tsx

```diff
@@ -60,38 +57,44 @@ export const useSearchMappedEquipmentAssetMappings = (
       ...models.map((model) => [model.modelId, model.revisionId])
     ],
     queryFn: async ({ pageParam }: { pageParam: string | undefined }) => {
-      if (initialAssetMappings === undefined) {
+      if (query === '' || assetMappingList === undefined) {
         return { assets: [], nextCursor: undefined };
       }
-      if (query === '') {
-        const assets = initialAssetMappings.pages.flatMap((modelWithAssets) =>
-          modelWithAssets.modelsAssets.flatMap((modelsAsset) => modelsAsset.assets).flat()
-        );
-        return { assets, nextCursor: undefined };
-      }
-      if (assetMappingList === undefined) {
-        return { assets: [], nextCursor: undefined };
-      }
-      const assetsResponse = await getAssetsList(sdk, {
-        query,
-        limit,
-        cursor: pageParam
-      });
 
-      const assets = assetsResponse.items.filter(isDefined);
-      const filteredSearchedAssets = assetMappingList.flatMap((mapping) => {
-        return mapping.assetMappings
-          .filter((assetMapping) => assets.some((asset) => asset.id === assetMapping.assetId))
-          .map((assetMapping) => assets.find((asset) => asset.id === assetMapping.assetId))
-          .filter(isDefined);
-      });
+      const fetchAssets = async (
+        cursor: string | undefined,
+        accumulatedAssets: Asset[]
+      ): Promise<{ assets: Asset[]; nextCursor: string | undefined }> => {
+        const assetsResponse = await getAssetsList(sdk, {
+          query,
+          limit,
+          cursor
+        });
 
-      // Remove duplicates
-      const uniqueFilteredSearchedAssets = uniq(filteredSearchedAssets);
+        const fetchedAssets = assetsResponse.items.filter(isDefined);
+        const filteredSearchedAssets = assetMappingList.flatMap((mapping) => {
+          return mapping.assetMappings
+            .filter((assetMapping) =>
+              fetchedAssets.some((asset) => asset.id === assetMapping.assetId)
+            )
+            .map((assetMapping) => fetchedAssets.find((asset) => asset.id === assetMapping.assetId))
+            .filter(isDefined);
+        });
+
+        const uniqueAssets = uniq([...accumulatedAssets, ...filteredSearchedAssets]);
```

Comment:
Are these two asset arrays containing assets within the same memory reference?
Correct if I'm wrong but I guess `uniq` doesnt do a deep comparison, only primitive types. If you use `uniqBy `to add some criteria for comparison than might fulfill your need to have unique assets, isnt?